comment.js

```diff
@@ -0,0 +1,26 @@
+// allow data to persist on page load
+(() => {
+  data = JSON.parse(localStorage.getItem("data")) || [];
```

Comment:
> Dont understand how the function thats allowing data to persist works. Specifically, the | | [ ] at the end of line 3 in comment.js

this could also be written as 
```js
const savedData = localStorage.getItem("data")
// ^^^ if "data" does not exist in localStorage, then this will be undefined
const maybeData = JSON.parse(savedData)
// ^^^ this could be your data object or undefined
if (maybeData !== undefined) {
  data = maybeData
} else {
  data = []
// ^^^ we set data to an empty array since there is no saved data
}
```

Reply:
ah much easier to read and understand. thanks

Reply:
it's also worth pointing out that `JSON.parse()` can throw an error. it's best practice to put it inside a try catch so that we can ..."catch" the error being "thrown"

Reply:
so that `|| []` is a declarative way of saying "js please grab this data from localstorage, oh and if it's falsey aka not there, just assign an empty array"